### PR TITLE
Cleanup: Add a utility to print collision pairs

### PR DESCRIPTION
**Pull request by rhaschke**
I think @tylerjw merged #2275 too early (without having approvals from two maintainers). This PR realizes some changes required in my opinion (see below).

### CODE REVIEW

File: moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h [hunk 1]
```diff
@@ -144,12 +144,7 @@ struct CostSource
 /** \brief Representation of a collision checking result */
 struct CollisionResult
 {
-  CollisionResult()
-    : collision(false)
-    , distance(std::numeric_limits<double>::max())
-    , contact_count(0)
-    , collision_log_period(5)
-    , collision_log_name("collision_common")
```
Comment (rhaschke) [inline moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h:152]: It doesn't make sense to add these constants as extra variables to a lightweight and often used struct like `CollisionResult`.

File: moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h [hunk 2]
```diff
@@ -166,25 +161,8 @@ struct CollisionResult
...
-      ROS_DEBUG_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name, "Objects in collision:");
-      for (auto contact : contacts)
-      {
-        ROS_DEBUG_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name,
-                                        "\t" << contact.first.first << ", " << contact.first.second);
-      }
-    }
-  }
+  /** \brief Throttled warning printing the first collision pair, if any. All collisions are logged at DEBUG level */
+  void print() const;
```
Comment (rhaschke) [inline moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h:165]: Moving the `print()` method into a .cpp file resolves the scope issue mentioned in https://github.com/ros-planning/moveit/pull/2275#discussion_r475567741.
As `print()` isn't lightweight nor performance-critical, it shouldn't be inline anyway.

File: moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h [hunk 3]
```diff
@@ -166,25 +161,8 @@ struct CollisionResult
...
-    if (!contacts.empty())
-    {
-      ROS_WARN_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name,
-                                     "Objects in collision (printing first collision pair of "
-                                         << contacts.size() << "): " << contacts.begin()->first.first << ", "
-                                         << contacts.begin()->first.second);
-
-      // Log all collisions at the debug level
-      ROS_DEBUG_STREAM_THROTTLE_NAMED(collision_log_period, collision_log_name, "Objects in collision:");
-      for (auto contact : contacts)
```
Comment (rhaschke) [inline moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h:181]: Here contact entries were copied for no reason. Strange, that clang-tidy didn't flag this issue.

File: moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h (hunk 2 above)
Comment (AndyZe) [reply moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h:165]: That sounds good to me. I was thinking about doing that but waiting for a nudge from a reviewer.

File: moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h (hunk 3 above)
Comment (AndyZe) [reply moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h:181]: thanks, I'll stop making this mistake now

**Pull request merged**